pkg/apis/operator/v1alpha1/tektonconfig_defaults.go

```diff
@@ -26,4 +26,20 @@ func (tc *TektonConfig) SetDefaults(ctx context.Context) {
 	}
 
 	tc.Spec.Pipeline.PipelineProperties.setDefaults()
+
+	// before webhook we had default value for pruner's keep as 1
+	// but we expect user to define all values now
+	// so if a user has installed prev version and has not enabled
+	// pruner then `keep` will have a value 1 and after upgrading
+	// to newer version webhook will fail if keep has a value and
+	// other fields are not defined
+	// this handles that case by removing the default for keep if
+	// other pruner fields are not defined
+	if len(tc.Spec.Pruner.Resources) == 0 {
```

Comment:
@pradeepitm12  if the neither `pipelinerun` or `taskrun` is there in the list, is that interpreted as prune both or prune none.